x/ccv/child/keeper/hooks.go

```diff
@@ -20,10 +20,16 @@ func (k Keeper) AfterValidatorDowntime(ctx sdk.Context, consAddr sdk.ConsAddress
 		return
 	}
 
+	// get last the valset update id
+	valsetUpdate, err := k.GetLastUnbondingPacketData(ctx)
+	if err != nil {
+		return
+	}
+
 	// increase jail time
 	signInfo.JailedUntil = ctx.BlockHeader().Time.Add(k.slashingKeeper.DowntimeJailDuration(ctx))
```

Comment:
Why does the consumer Slashing module need the `JailedUntil` to be set? AFAIK, it is only used to check whether a validator can unjail itself (i.e., handling a `MsgUnjail`). However, the consumer chain will have `MsgUnjail` disabled. 

If we need a variable `JailedUntil` for the CCV logic, then I'd keep it internal to CCV.

Reply:
Agreed

Reply:
> Why does the consumer Slashing module need the JailedUntil to be set?

To know when it has already sent a slashing packet for a distinct validator. Otherwise, if the validator is still down during the consecutive blocks, the consumer would spam the provider with slashing packets. What would be useless and expensive.

Reply:
But the slashing module doesn't send slashing packets. The CCV module does.

Reply:
It seems to me that it would be pretty complicated and involve redundant state to have the CCV module keep track of the missed blocks and jailedUntil. I think that we should keep modifying this state, but do it in a neater way and not use the SetValidatorSigningInfo setter directly, instead having an method on slashing to ResetMissedBlocks that would handle the internal logic of the missed block array and associated variables. Then addressing JailedUntil, there already is a JailUntil method that sets that directly that can be used.

Reply:
>It seems to me that it would be pretty complicated and involve redundant state to have the CCV module keep track of the missed blocks and jailedUntil.

I don't think we should do that either. The consumer slashing module should keep doing the slashing logic (i.e., count the missing blocks and trigger a slash). Triggering a slash entails telling the consumer CCV module to send a SlashPacket to the provider CCV module. 

The `jailedUntil` state on the slashing module is not used on the **_consumer_** chain. It's used in the single-chain case to decide whether a validator can be unjailed. The `jailedUntil` that we use on the consumer chain is part of the CCV module logic, i.e., to avoid sending multiple SlashPackets for the same infraction (in case of downtime). What I'm saying is that **_if it is CCV module logic then it should be part of the CCV module state_**.